src/odemis/gui/comp/overlay/rectangle.py

```diff
@@ -88,18 +138,83 @@ def restore_state(self, state: RectangleState):
         self._points = self.get_physical_sel()
         self.points.value = self._points
 
-    def is_point_in_shape(self, point):
+    def is_point_in_shape(self, v_point):
         if self._points:
-            return point_in_polygon(point, self._points)
+            hover, _ = self.get_hover(v_point)
```

Comment:
This change is odd. Firstly, `get_hover()` does extra things, and then calls `point_in_polygon()`, so it can only be slower than before. While this PR is about making things faster. Also, `get_hover()` will return true for all the "edges", including the rotation knob, which is can be outside the shape. So now the function will return True even if the point is not within the shape. Either, this is what you want, and you should rename the function (and add a docstring), or it's not what you want, and actually should use only `point_in_polygon()`.

Reply:
This is what you want, I will rename the function and add the docstring